pkg/resource/job_run/hooks.go

```
@@ -0,0 +1,147 @@
+package job_run
+
+import (
+	//"encoding/json"
+	// "fmt"
+
+	"reflect"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	"github.com/aws/aws-sdk-go/aws"
+	svcsdk "github.com/aws/aws-sdk-go/service/emrcontainers"
+	"github.com/ghodss/yaml"
+)
+
+func cfgToString(cfg *svcsdk.ConfigurationOverrides) (*string, error) {
+	configBytes, err := yaml.Marshal(cfg)
+	if err != nil {
+		return nil, err
+	}
+	configStr := string(configBytes)
+	return &configStr, nil
+}
+
+func stringToConfigurationOverrides(cfg *string) (*svcsdk.ConfigurationOverrides, error) {
+	if cfg == nil {
+		cfg = aws.String("")
+	}
+
+	var config svcsdk.ConfigurationOverrides
+	err := yaml.Unmarshal([]byte(*cfg), &config)
+	if err != nil {
+		return nil, err
+	}
+	return &config, nil
+}
+
+func customPreCompare(
+	delta *ackcompare.Delta,
+	a *resource,
+	b *resource,
+) {
+	aConfig, err := stringToConfigurationOverrides(a.ko.Spec.ConfigurationOverrides)
+	if err != nil {
+		panic(err)
```

Panic-ing seems risky. Since it would break reconciliation for all the other resources handled by this controller. I understand we don't return errors in this method, but maybe we could just force it to return a delta if we find an error?

---

Ok so yeah panicking is not really an option. I just did some brainstorming with Amine about this. We think it would be best to validate the user input in a hook before the create request. Something that just attempts to parse it from YAML to the Go type. If that fails, fail early with a terminal condition error. After that, it should be created as per usual. Then whatever we know is being returned by the server has to be valid, and we can essentially ignore these errors. (Maybe we still want to debug log it, just in case this ever changes)

---

Hi @RedbackThomson , we are doing sanity checks here in this [pre-create hook](https://github.com/aws-controllers-k8s/emrcontainers-controller/pull/21/files#diff-9baa4732154f72d148c3a43ded927cb11cd18611fe1f1a378d80aa91ccef5abf). Let me know if this is enough